### PR TITLE
New package py-picrust2 version 2.3.0-b with py-biom-format dep. fixed

### PR DESCRIPTION
**Pull request by dorton21**

### CODE REVIEW

**Comment by dorton21** [comment]
Can I get an update regarding the ability of this pull request to go through?

**Review by scheibelp — APPROVED** [review]

**Comment by scheibelp** [comment]
~Sorry the PR can be merged without the pipeline reporting, I have approved it and will merge once CI completes.~ Thanks for the ping!

(EDIT: looks like Adam has some suggestions which would improve this)

File: var/spack/repos/builtin/packages/py-biom-format/package.py [hunk 1]
```diff
@@ -14,17 +14,20 @@ class PyBiomFormat(PythonPackage):
...
     version('2.1.6', sha256='8eefc275a85cc937f6d6f408d91b7b45eae854cd5d1cbda411a3af51f5b49b0d')
 
     variant('h5py', default=True, description='For use with BIOM 2.0+ files')
 
     depends_on('py-setuptools', type=('build', 'run'))
     depends_on('py-cython', type='build')
     depends_on('py-h5py', type=('build', 'run'), when='+h5py')
-    depends_on('py-click', type=('build', 'run'))
+    depends_on('py-click', type=('build', 'run'), when='@2.1.5:')
     depends_on('py-numpy@1.3.0:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biom-format/package.py]:
```suggestion
    depends_on('py-numpy@1.9.2:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-biom-format/package.py [hunk 2]
```diff
@@ -14,17 +14,20 @@ class PyBiomFormat(PythonPackage):
...
     depends_on('py-future@0.16.0:', type=('build', 'run'))
     depends_on('py-scipy@0.13.0:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biom-format/package.py]:
```suggestion
    depends_on('py-scipy@1.3.1:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-biom-format/package.py [hunk 3]
```diff
@@ -14,17 +14,20 @@ class PyBiomFormat(PythonPackage):
...
     depends_on('py-pandas@0.19.2:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biom-format/package.py]:
```suggestion
    depends_on('py-pandas@0.20.0:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-biom-format/package.py [hunk 4]
```diff
@@ -14,17 +14,20 @@ class PyBiomFormat(PythonPackage):
...
 
+    version('2.1.10', sha256='f5a277a8144f0b114606852c42f657b9cfde44b3cefa0b2638ab1c1d5e1d0488')
+    version('2.1.9', sha256='18a6e4d4b4b2a6bf2d5544fa357ad168bedeac93f0837015ef9c72f41fa89491')
+    version('2.1.7', sha256='b47e54282ef13cddffdb00aea9183a87175a2372c91a915259086a3f444c42f4')
     version('2.1.6', sha256='8eefc275a85cc937f6d6f408d91b7b45eae854cd5d1cbda411a3af51f5b49b0d')
 
     variant('h5py', default=True, description='For use with BIOM 2.0+ files')
 
     depends_on('py-setuptools', type=('build', 'run'))
     depends_on('py-cython', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biom-format/package.py]:
```suggestion
    depends_on('py-cython@0.29:', type='build')
```

File: var/spack/repos/builtin/packages/py-biom-format/package.py [hunk 5]
```diff
@@ -14,17 +14,20 @@ class PyBiomFormat(PythonPackage):
...
     depends_on('py-h5py', type=('build', 'run'), when='+h5py')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biom-format/package.py]: Seems like this is now required?

File: var/spack/repos/builtin/packages/py-biom-format/package.py [hunk 6]
```diff
@@ -14,17 +14,20 @@ class PyBiomFormat(PythonPackage):
     homepage = "https://pypi.python.org/pypi/biom-format/2.1.6"
     url      = "https://pypi.io/packages/source/b/biom-format/biom-format-2.1.6.tar.gz"
 
+    version('2.1.10', sha256='f5a277a8144f0b114606852c42f657b9cfde44b3cefa0b2638ab1c1d5e1d0488')
+    version('2.1.9', sha256='18a6e4d4b4b2a6bf2d5544fa357ad168bedeac93f0837015ef9c72f41fa89491')
+    version('2.1.7', sha256='b47e54282ef13cddffdb00aea9183a87175a2372c91a915259086a3f444c42f4')
     version('2.1.6', sha256='8eefc275a85cc937f6d6f408d91b7b45eae854cd5d1cbda411a3af51f5b49b0d')
 
     variant('h5py', default=True, description='For use with BIOM 2.0+ files')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biom-format/package.py]: Add this:
```python
depends_on('python@2.7:')
depends_on('python@3:', when='@2.1.9:')
```

File: var/spack/repos/builtin/packages/py-biom-format/package.py [hunk 7]
```diff
@@ -14,17 +14,20 @@ class PyBiomFormat(PythonPackage):
...
     depends_on('py-h5py', type=('build', 'run'), when='+h5py')
-    depends_on('py-click', type=('build', 'run'))
+    depends_on('py-click', type=('build', 'run'), when='@2.1.5:')
     depends_on('py-numpy@1.3.0:', type=('build', 'run'))
     depends_on('py-future@0.16.0:', type=('build', 'run'))
     depends_on('py-scipy@0.13.0:', type=('build', 'run'))
     depends_on('py-pandas@0.19.2:', type=('build', 'run'))
     depends_on('py-six@1.10.0:', type=('build', 'run'))
-    depends_on('py-pyqi', type=('build', 'run'))
+    depends_on('py-pyqi', type=('build', 'run'), when='@:2.1.4')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biom-format/package.py]:
```suggestion
    depends_on('py-pyqi', type=('build', 'run'), when='^python@:2')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-biom-format/package.py [hunk 8]
```diff
@@ -14,17 +14,22 @@ class PyBiomFormat(PythonPackage):
...
 
+    version('2.1.10', sha256='f5a277a8144f0b114606852c42f657b9cfde44b3cefa0b2638ab1c1d5e1d0488')
+    version('2.1.9', sha256='18a6e4d4b4b2a6bf2d5544fa357ad168bedeac93f0837015ef9c72f41fa89491')
+    version('2.1.7', sha256='b47e54282ef13cddffdb00aea9183a87175a2372c91a915259086a3f444c42f4')
     version('2.1.6', sha256='8eefc275a85cc937f6d6f408d91b7b45eae854cd5d1cbda411a3af51f5b49b0d')
 
     variant('h5py', default=True, description='For use with BIOM 2.0+ files')
 
+    depends_on('python@2.7:')
+    depends_on('python@3:', when='@2.1.9:')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-biom-format/package.py]:
```suggestion
    depends_on('python@2.7:', type=('build', 'run'))
    depends_on('python@3:', when='@2.1.9:', type=('build', 'run'))
```

**Comment by dorton21** [comment]
Any other fixes needed for this?

File: var/spack/repos/builtin/packages/py-picrust2/package.py [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+    version('2.3.0-b', sha256='ac12c372bc263e750d9101eca0cd0e57de37089b661fa1a13caf5a544d293737')
+    version('2.2.0-b', sha256='c41e1f487b33179f4aecede50cfd8b652aa3cef2ea1ae5fd022f531c7d549097')
+    version('2.1.4-b', sha256='f781eb323914979b6d3bca088a5152f085f53e6e38f1c3be94b35f99fc1db2d8')
+
+    depends_on('py-setuptools', type=('build'))
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('py-h5py', type=('build', 'run'))
+    depends_on('py-joblib', type=('build', 'run'))
+    depends_on('py-biom-format@2.1.5:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-picrust2/package.py]: Where do you see that 2.1.5+ is needed?

File: var/spack/repos/builtin/packages/py-biom-format/package.py (hunk 5 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-biom-format/package.py]: Since this is now required, we might want to remove the variant.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-biom-format/package.py]: You can also remove the when arg.

**Review by adamjstewart — APPROVED** [review]

**Comment by dorton21** [comment]
What is the unit test failure?

**Comment by adamjstewart** [comment]
Looks like a transient failure, I restarted the tests.

**Review by alalazo — APPROVED** [review]

**Pull request merged**